ai_server/server.py

```diff
@@ -55,14 +55,19 @@ def chat_with_llama_server_http(
     timeout: int = 300,
     image_files: Optional[list] = None,
     json_schema: Optional[dict] = None,
+    model_options: Optional[dict] = None,
 ) -> str:
     """Handle chat using llama-server HTTP API."""
     if not LLAMA_SERVER_URL:
         raise Exception("LLAMA_SERVER_URL environment variable not set")
 
     try:
         messages = _build_messages(content, system_prompt, image_files=[])  # TODO: Pass image files
-        payload = {'model': model, 'messages': messages, 'stream': False, 'max_tokens': 512}
+
+        if not model_options:
+            model_options = {}
+
+        payload = {'model': model, 'messages': messages, **model_options}
```

Comment:
Please keep the `stream` option hard coded. We can allow the client to override the default value, but we should set a False default.